src/tribler-gui/tribler_gui/widgets/channelcontentswidget.py

```diff
@@ -268,7 +270,13 @@ def go_back(self, checked=False):
     def on_breadcrumb_clicked(self, tgt_level):
         if int(tgt_level) + 1 != len(self.channels_stack):
             self.go_back_to_level(tgt_level)
+        elif isinstance(self.model, SearchResultsModel) and len(self.channels_stack) == 1:
+            # In case of remote search, when only the search results are on the stack,
+            # we must keep the txt_filter (which contains the search term) before resetting the view
+            text_filter = self.model.text_filter
+            self.reset_view(text_filter=text_filter)
         else:
+            # Reset the view if the user clicks on the last part of the breadcrumb
```

Comment:
**issue:** I still disagree with resetting the view by clicking on the rightmost breadcrumb 
More information: https://github.com/Tribler/tribler/pull/5876#issuecomment-750223950
**suggestion:** remove reset logic from click on rightmost breadcrumb or implement the more straightforward general solution.

Reply:
> suggestion: remove reset logic from click on rightmost breadcrumb or implement the more straightforward general solution.

IMHO, adding a special "Reset" button to the GUI - that would be overkill. The fewer buttons - the better. Also, the current solution is general enough (but, sadly, not too straightforward.)

Reply:
You can add reset logic to the leftmost breadcrumb.

Reply:
The leftmost breadcrumb is for navigation to the root of the channels stack...

Reply:
> The leftmost breadcrumb is for navigation to the root of the channels stack...

From my understanding, it is the same as "reset". Am I wrong?

Reply:
Clicking on each part of the path in the breadcrumb will get you to the corresponding part of the channels stack.

When you're looking at the contents of folder `D` the breadcrumb looks like this:
`A / B / C / D`

Clickin on "B" will pop `D` and `C` from the stack, gettin you to look at the contents of `B`. That is straightforward and intuitive behaviour. Clicking on `A` will show you the contents of `A` (A could be either a search request root, a channel or a list of discovered channels). Clicking on `D` (the name of the current folder in the breadcrumb) will reset the contents of `D`.